Sources/SpeziOnboarding/OnboardingFlow/OnboardingNavigationPath.swift

```diff
@@ -97,30 +120,180 @@ public class OnboardingNavigationPath {
+            // Without this limitation, attempts to navigate backwards or dismiss the currently displayed onboarding step
+            // (for example, after receiving HealthKit authorizations) could lead to unintended behavior.
+            let currentStepIndex = currentOnboardingStep.flatMap {
+                onboardingSteps.elements.keys.firstIndex(of: $0)
+            } ?? 0
+            
+            // Remove all onboarding steps after the current onboarding step
+            let nextStepIndex = currentStepIndex + 1
+            if nextStepIndex < onboardingSteps.elements.endIndex {
+                onboardingSteps.removeSubrange(nextStepIndex...)
+            }
+            
+            for (elementIdx, element) in elements.enumerated() {
+                let onboardingStepIdentifier = OnboardingStepIdentifier(element: element)
+                let stepIsAfterCurrentStep = elementIdx > currentStepIndex // !self.onboardingSteps.keys.contains(onboardingStepIdentifier)
+                guard stepIsAfterCurrentStep else {
+                    continue
+                }
+                
+                guard self.onboardingSteps[onboardingStepIdentifier] == nil else {
+                    preconditionFailure("""
+                    SpeziOnboarding: Duplicate Onboarding step identifier hash `\(onboardingStepIdentifier)` identified.
+                    Ensure unique Onboarding view identifiers within the `OnboardingStack`!
+                    """)
+                }
+                
+                self.onboardingSteps[onboardingStepIdentifier] = element.view
+            }
+        }
+        onboardingComplete()
+    }
+    
+    
+    private func onboardingComplete() {
+        if self.onboardingSteps.isEmpty && !(self.isComplete?.wrappedValue ?? false) {
+            self.isComplete?.wrappedValue = true
+        }
+    }
+}
 
 
+// MARK: Navigation
+
+extension OnboardingNavigationPath {
+    /// Internal function used to navigate to the respective onboarding `View` via the `NavigationStack.navigationDestination(for:)`,
+    /// either regularly declared within the ``OnboardingStack`` or custom steps
+    /// passed via ``append(customView:)``, identified by the `OnboardingStepIdentifier`.
+    ///
+    /// - Parameters:
+    ///   - stepIdentifier: The onboarding step identified via `OnboardingStepIdentifier`
+    /// - Returns: `View` corresponding to the passed `OnboardingStepIdentifier`
+    func view(for stepIdentifier: OnboardingStepIdentifier) -> AnyView {
+        if stepIdentifier.isCustom {
+            guard let view = customOnboardingSteps[stepIdentifier] else {
+                return AnyView(IllegalOnboardingStepView())
+            }
+            return AnyView(view)
+        }
+        guard let view = onboardingSteps[stepIdentifier] else {
+            return AnyView(IllegalOnboardingStepView())
+        }
+        return AnyView(view)
+    }
```

Comment:
I guess it's rather hard to remove the AnyView type erasure here? I think I couldn't make it work back then, you might have some more ideas

Reply:
i thought about this as well, bit since the values stored in `customOnboardingSteps` are `any View` existentials, we need to wrap it AFAICT